src/core/tranche/orchestrator.sol

```diff
@@ -0,0 +1,42 @@
+// Copyright (C) 2019 Centrifuge
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.4.24;
+
+import "ds-note/note.sol";
+
+import {Diffuser} from "./diffuser.sol";
+import {Distributor} from "./distributor.sol";
+
+contract Orchestrator is DSNote{
+
+    Diffuser public diffuser;
+    Distributor public distributor;
```

Comment:
If we are implementing the Orchestrator. 
@ilinzweilin  might be right and we never need both in one deployment.

I would use here a `DistributorLike` abstract contract/interface